shim.js

```diff
@@ -16,6 +16,11 @@ var define = function defineGetter(getter) {
 	});
 };
 
+var copyStaticProperties = function copyStatProperties(ctorNew, ctorOld) {
+	setProto(ctorNew, Object.getPrototypeOf(ctorOld));
+	Object.defineProperties(ctorNew, Object.getOwnPropertyDescriptors(ctorOld));
```

Comment:
- `Object.getPrototypeOf` may not exist in every engine, and either way, it needs to be cached at module level
 - `Object.defineProperties` needs to be cached at module level
 - `Object.getOwnPropertyDescriptors` should instead use https://www.npmjs.com/package/object.getownpropertydescriptors

Reply:
cache is not needed, shim will run only once

Reply:
Caching is always needed; and no, the shim method can run many times. The file will only be *required* once.